server/ServerFSM.cpp

```diff
@@ -456,6 +468,19 @@ bool ServerFSM::EstablishPlayer(const PlayerConnectionPtr& player_connection,
             client_type == Networking::CLIENT_TYPE_HUMAN_OBSERVER ||
             client_type == Networking::CLIENT_TYPE_HUMAN_PLAYER)
         {
+            // add "player enter game" message
+            boost::posix_time::ptime timestamp = boost::posix_time::second_clock::universal_time();
+            std::string data = player_name + " [[" + UserStringNop("PLAYER_ENTERED_GAME") + "]]";
```

Comment:
It translates `PLAYER_ENTERED_GAME` when the chatbox sees the `[[` and `]]`, but this is doing the string composition in code with `+` instead of substitution with `FlexibleFormat`, which would allow the stringtable / translation to control where and how the name appears in the string.

Reply:
Changed to boost::format

Reply:
You missed the part about using the stringtable entry to determine where and how the name appears in the string. Just using `boost::format` instead of `+` but keeping the position of the name in the string the same is worse than it was before.